crates/anvil/src/eth/api.rs

```diff
@@ -1351,7 +1351,26 @@ impl EthApi {
     /// Handler for ETH RPC call: `eth_maxPriorityFeePerGas`
     pub fn max_priority_fee_per_gas(&self) -> Result<U256> {
         node_info!("eth_maxPriorityFeePerGas");
-        Ok(U256::from(self.backend.max_priority_fee_per_gas()))
+        Ok(U256::from(self.lowest_suggestion_tip().unwrap_or(1e9 as u128)))
+    }
+
+    /// Returns the suggested fee cap
+    fn lowest_suggestion_tip(&self) -> Option<u128> {
+        let fee_history_cache = {
+            let history = self.fee_history_cache.lock();
+            history.clone()
+        };
```

Comment:
```suggestion
        let fee_history_cache = self.fee_history_cache.lock();
```

Reply:
I was asked to clone the history and drop the lock, as far as I understood

Reply:
I see, I think that's not necessary.

@mattsse that doesn't seem right, if anything it should be made into a rwlock.

Reply:
done